handlers/mparticle-api/src/index.ts

```
@@ -1,164 +1,84 @@
-                    eventType: z.string(),
-                })),
-                deviceInfo: z.record(z.string(), z.unknown()),
-                userAttributes: z.record(z.string(), z.unknown()),
-                deletedUserAttributes: z.array(z.string()),
-                userIdentities: z.record(z.string(), z.unknown()),
-                applicationInfo: z.record(z.string(), z.unknown()),
-                schemaVersion: z.number(),
-                environment: z.enum(['production', 'development']),
-                context: z.record(z.string(), z.unknown()),
-                ip: z.string(),
-            })
-        }
-    })
-]);
+// Baton RER event handler function
+async function handleBatonRerEvent(
+	event: BatonRerEventRequest,
+	context: Context,
+): Promise<BatonRerEventResponse> {
+	try {
+		return batonRerRouter.routeRequest(event);
+	} catch (error) {
+		console.error('Baton RER handler error:', error);
+		throw error; // Re-throw to trigger Lambda retry mechanism
+	}
+}
 
 export const handler: Handler = async (
-    event: APIGatewayProxyEvent,
-): Promise<APIGatewayProxyResult> => {
-    return router.routeRequest(event);
+	event: APIGatewayProxyEvent | BatonRerEventRequest | any,
+	context: Context,
+): Promise<APIGatewayProxyResult | BatonRerEventResponse> => {
+	if (isAPIGatewayEvent(event)) {
+		console.debug('Processing as API Gateway event');
+		return handleHttpRequest(event, context);
+	} else if (isBatonRerEvent(event)) {
+		console.debug('Processing as Baton RER event');
+		return handleBatonRerEvent(event, context);
+	} else {
+		// Handle other event types or unknown events
+		console.error('Unknown event type:', event);
+
+		// If it might be an HTTP event without proper structure, try to handle gracefully
+		if (event.httpMethod || event.path) {
+			console.log('Attempted to process as malformed API Gateway event');
+			return {
+				statusCode: 400,
+				body: JSON.stringify({ error: 'Malformed API Gateway event' }),
+			};
+		}
```

make sure it returns a 500 or throws an exception as we want to receive an alarm in this case either from failed lambda or 5xx error.  I would personally just remove this block completely and let it throw an exception if there's any issue with the request.